intelmq/app/server.py

```diff
@@ -0,0 +1,96 @@
+"""Main entrypoint for the IntelMQ server application
+
+SPDX-FileCopyrightText: 2022 CERT.at GmbH <https://cert.at>, 2023 Filip Pokorný
+SPDX-License-Identifier: AGPL-3.0-or-later
+"""
+
+import os
+import gunicorn.app.base
+import uvicorn
+import pathlib
+
+from fastapi import FastAPI
+from fastapi.staticfiles import StaticFiles
+from fastapi.middleware.cors import CORSMiddleware
+from uvicorn.workers import UvicornH11Worker
+
+from intelmq.version import __version__
+import intelmq.app.dependencies
+import intelmq.app.api.exceptions
+
+from intelmq.app.config import Config
+from intelmq.app.api.router import router as api_router
+from intelmq.app.webgui.router import router as web_router
+
+
+class IntelMQUvicornWorker(UvicornH11Worker):
+    pass
+
+
+class IntelMQGunicornApplication(gunicorn.app.base.BaseApplication):
+
+    def __init__(self, application, options=None):
+        self.options = options or {}
+        self.application = application
+        super().__init__()
+
+    def load_config(self):
+        cfg = {key: value for key, value in self.options.items()
+                  if key in self.cfg.settings and value is not None}
+        for key, value in cfg.items():
+            self.cfg.set(key.lower(), value)
+
+    def load(self):
+        return self.application
+
+
+config = Config()
+
+app = FastAPI(
+    title="IntelMQ",
+    version=__version__,
+    root_path=os.environ.get("ROOT_PATH", ""),
+)
+
+
+@app.on_event("startup")
+def init_app():
+    intelmq.app.dependencies.startup(config)
+
+
+app.add_middleware(CORSMiddleware, allow_origins=config.allow_origins, allow_methods=("GET", "POST"))
+app.include_router(api_router, prefix="/api/v1")
```

Comment:
My argument is that while technically the API is publicly available, it is not public per se, currently the endpoints are not documented for the outside use nor do we explicitly say "this is a public API, feel free to use it". As it stands right now it is more of an internal API used for the Manager only. As such I don't think we need to "support" it for other usage.
It's kind of like if you reverse engineer an internal API of some sort, you can't expect the developer to keep you in mind and don't change the API because of you.